src/main/java/org/gridsuite/modification/server/modifications/GenerationDispatch.java

```diff
@@ -522,13 +523,65 @@ public void apply(Network network, Reporter subReporter) {
                     Map.of(), TypedValue.INFO_SEVERITY);
+                generatorsByRegion.forEach((region, generators) -> report(resultReporter, Integer.toString(componentNum), "SumGeneratorActivePower" + region, "Sum of generator active power setpoints in ${region} region: ${sum} MW.",
+                        Map.of("region", region, "sum", getSumActivePower(generators)), TypedValue.INFO_SEVERITY));
             } else {
                 double remainingPowerImbalance = totalAmountSupplyToBeDispatched - realized;
                 report(resultReporter, Integer.toString(componentNum), "SupplyDemandBalanceCouldNotBeMet", "The supply-demand balance could not be met : the remaining power imbalance is ${remainingPower} MW",
                     Map.of("remainingPower", remainingPowerImbalance), TypedValue.WARN_SEVERITY);
             }
         }
     }
+
+    private Map<String, List<Generator>> getGeneratorsByRegion(Network network) {
+        // get all connected generators and the substationIds associated.
+        List<Generator> connectedGenerators = network.getGeneratorStream()
+                .filter(g -> g.getTerminal().isConnected())
+                .toList();
+        List<String> substationIds = connectedGenerators.stream()
+                .map(g -> g.getTerminal().getVoltageLevel().getSubstation().orElseThrow().getId())
+                .toList();
+        // get all substations with "regionCvg" property name
+        Map<String, String> substationIdPropertiesMap = new HashMap<>();
+        if (!CollectionUtils.isEmpty(substationIds)) {
+            substationIds.forEach(sId -> {
+                Substation substation = network.getSubstation(sId);
+                if (!substation.getPropertyNames().isEmpty() && hasCvgPropertyName(substation.getPropertyNames())) {
+                    substation.getPropertyNames().forEach(property -> {
+                        if (REGION_CVG.equals(property)) {
+                            substationIdPropertiesMap.put(substation.getId(), substation.getProperty(property));
+                        }
+                    });
+                }
+            });
+        }
+
+        // group substationIds by region
+        Map<String, List<String>> groupedSubstationIds = substationIdPropertiesMap.keySet().stream().collect(Collectors.groupingBy(substationIdPropertiesMap::get));
+
+        // iterate over groupedSubstationIds and check for each substation list if it's related to the connected generators
+        Map<String, List<Generator>> generatorsByRegion = new HashMap<>();
+
+        groupedSubstationIds.forEach((region, substationList) -> {
+            List<Generator> connectedGeneratorsWithSubstation = connectedGenerators.stream()
+                    .filter(g -> substationList.contains(g.getTerminal().getVoltageLevel().getSubstation().orElseThrow().getId()))
```

Comment:
Same : Don't throw an exception for a voltageLevel not in a substation, which can occur : just ignore the generator in that case

Reply:
Done.